index.d.ts

```diff
@@ -74,20 +74,32 @@ declare module '@primer/components' {
 
   export const Heading: React.FunctionComponent<HeadingProps>
 
-  type DetailsRenderFunction = (args: {open: boolean}) => React.ReactElement
 
   export interface DetailsProps extends CommonProps, Omit<React.DetailsHTMLAttributes<HTMLDetailsElement>, 'color'> {
-    render?: DetailsRenderFunction
-    children?: DetailsRenderFunction | React.ReactNode
-    defaultOpen?: boolean
-    overlay?: boolean
-    open?: boolean
-    onToggle?: (event: React.SyntheticEvent<HTMLDetailsElement>) => void
-    onClickOutside?: (event: MouseEvent) => void
+    onToggle: (event: React.SyntheticEvent<HTMLDetailsElement>) => void
+    open: boolean
+    ref: React.RefObject<HTMLDetailsElement>
   }
 
   export const Details: React.FunctionComponent<DetailsProps>
 
+  export interface UseDetailsProps {
+    defaultOpen?: boolean
+    closeOnOutsideClick?: boolean
+    onClickOutside?: (event: React.MouseEvent ) => void
+    ref?: React.RefObject<HTMLDetailsElement> | null
+  }
+
+  export const useDetails: (props: UseDetailsProps) => {
```

Comment:
I think the `props` here can be optional.
```suggestion
  export const useDetails: (props?: UseDetailsProps) => {
```